src/main/java/org/apache/hadoop/fs/PrestoFileSystemCache.java

```diff
@@ -54,13 +55,21 @@
     FileSystem get(URI uri, Configuration conf)
             throws IOException
     {
+        if (conf instanceof FileSystemFactory) {
```

Comment:
It looks like a hack. Can you please explain why it is done this way? What is the goal of this change?

Reply:
This is a hook to allow creating a file system object every time by passing in a custom `Configuration` object that implements `FileSystemFactory`. We use this for an internal file system, but would also be useful for S3. It would let us remove all the configuration bridging code and the hack to export stats, since the file system would be a wrapper around a client object that is lifecycled normally with Guice.

Reply:
No, because the cache is a singleton, and that would require making the factory a singleton, which we don't want.

Reply:
Fair enough. Please add those details to the commit message / code comments for the benefit of future readers.

Reply:
Thanks